ksql-engine/src/main/java/io/confluent/ksql/schema/registry/KsqlSchemaRegistryClientFactory.java

```diff
@@ -35,23 +37,41 @@
 
   private final SslFactory sslFactory;
   private final Supplier<RestService> serviceSupplier;
+  private final Map<String, Object> schemaRegistryClientConfigs;
+  private final SchemaRegistryClientFactory schemaRegistryClientFactory;
+
+  interface SchemaRegistryClientFactory {
+    CachedSchemaRegistryClient create(RestService service, int identityMapCapacity,
+                                      Map<String, Object> clientConfigs);
+  }
+
 
   public KsqlSchemaRegistryClientFactory(final KsqlConfig config) {
-    this(config, () -> new RestService(config.getString(KsqlConfig.SCHEMA_REGISTRY_URL_PROPERTY)),
-         new SslFactory(Mode.CLIENT));
+    this(config,
+        () -> new RestService(config.getString(KsqlConfig.SCHEMA_REGISTRY_URL_PROPERTY)),
+        new SslFactory(Mode.CLIENT),
+        (service, mapCapacity, clientConfigs) -> new CachedSchemaRegistryClient(service,
+                                                                                mapCapacity,
+                                                                                clientConfigs)
+    );
 
     // Force config exception now:
     config.getString(KsqlConfig.SCHEMA_REGISTRY_URL_PROPERTY);
   }
 
   KsqlSchemaRegistryClientFactory(final KsqlConfig config,
                                   final Supplier<RestService> serviceSupplier,
-                                  final SslFactory sslFactory) {
+                                  final SslFactory sslFactory,
+                                  final SchemaRegistryClientFactory schemaRegistryClientFactory) {
     this.sslFactory = sslFactory;
     this.serviceSupplier = serviceSupplier;
+    this.schemaRegistryClientConfigs = config.originalsWithPrefix(
```

Comment:
The constructor now calls `config.originalsWithPrefix(KsqlConfig.KSQL_SCHEMA_REGISTRY_PREFIX)` twice. You could of just passed `schemaRegistryClientConfigs` to `sslFactory.configure` :)